stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -1249,7 +1257,12 @@ void DiscardingTaskGroup::offer(AsyncTask *completedTask, AsyncContext *context)
                                        alreadyDecrementedStatus);
             break;
           case ReadyStatus::Error:
-            SWIFT_TASK_GROUP_DEBUG_LOG(this, "offer, complete, resume with errorItem.task:%p", readyErrorItem.getTask());
+            // The completed task failed, but we already stored a different failed task.
+            // Thus we discard this error and complete with the previously stored.
+            SWIFT_TASK_GROUP_DEBUG_LOG(this, "offer, complete, discard error completedTask %p, resume with errorItem.task:%p",
+                                       completedTask,
+                                       readyErrorItem.getTask());
+            _swift_taskGroup_detachChild(asAbstract(this), completedTask);
```

Comment:
the fix™

We missed to detach this task and just continued to resume with the `readyErrorItem.getTask()`